docs/core/tools/dotnet-publish.md

```diff
@@ -153,13 +153,13 @@ For more information, see the following resources:
 
   For more information about single-file publishing, see the [single-file bundler design document](https://github.com/dotnet/designs/blob/main/accepted/2020/single-file/design.md).
 
-  We recommend that you specify this option in a publish profile rather than on the command line. For more information, see [MSBuild](#msbuild).
+  We recommend that you specify this option in the project file rather than on the command line. Setting it in the project file enables a Roslyn analyzer that provides single-file compatibility warnings during `dotnet build`. For more information, see [MSBuild](#msbuild).
 
 - **`-p:PublishTrimmed=true`**
```

Comment:
```suggestion
- **`<PublishTrimmed>true</PublishTrimmed>`**
```

Reply:
Same for `PublishSingleFile` above

Reply:
Unfortunately this document is talking about options for the `dotnet publish` command. Should we maybe just remove these sections from this file? It's strange that they're documented as publish options when really they're just passed to MSBuild. And these same docs then recommend _not_ using the command-line arg version.

Reply:
Removing them seems best, although the unfortunate benefit is that if you specific `PublishTrimmed` on the command line then it applies to all projects, and would turn on the analyzer for all of them. I suppose we also need to recommend adding `EnableTrimmingAnalyzer` in every project.

Reply:
I removed PublishTrimmed, PublishSingleFile, and PublishReadyToRun. @tdykstra does this seem ok to you?